smcore/factory.cpp

```diff
@@ -22,19 +22,32 @@
 
 #include "factory.h"
 
+#include <map>
+
 namespace smos
 {
     namespace smcore
     {
         //******************************************************************************
         Options *Factory::getOptions(smos::smcore::SMString optionsfile)
         {
-            static Options *smOptions = nullptr;
-            if (nullptr == smOptions)
+            static std::map<smos::smcore::SMString, Options *> programOptions;
```

Comment:
If you don't care about the order of objects in the map, you can use std::unordered_map which is faster

Reply:
Good hint. I was thinking about this, but I've expected not so much various config files to keep, so it seems to not matter. But I've changed this :D